plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2585,7 +2585,13 @@ protected DevicesDef createDevicesDef(VirtualMachineTO vmTO, GuestDef guest, int
             Map<String, String> details = vmTO.getDetails();
 
             boolean isIothreadsEnabled = details != null && details.containsKey(VmDetailConstants.IOTHREADS);
-            devices.addDevice(createSCSIDef(vcpus, isIothreadsEnabled));
+            int controllers = vmTO.getDisks().length / 7;
+            if (vmTO.getDisks().length % 7 != 0) {
+                controllers++;
+            }
+            for (int i = 0; i < controllers; i++) {
+                devices.addDevice(createSCSIDef((short)i, vcpus, isIothreadsEnabled));
+            }
```

Comment:
can this be a separate method?

Reply:
@DaanHoogland 
sure if you wish. I will add a method

Reply:
created a method, thanks for review @DaanHoogland